src/blob.rs

```diff
@@ -68,10 +69,67 @@ impl<'repo> Drop for Blob<'repo> {
     }
 }
 
+/// A structure to represent a git writestream for blobs
+pub struct BlobWriter<'repo> {
+    raw: *mut raw::git_writestream,
+    need_cleanup: bool,
+    _marker: marker::PhantomData<Object<'repo>>,
+}
+
+impl<'repo> BlobWriter<'repo> {
+    /// Finalize blob writing stream and write the blob to the object db
+    pub fn commit(&mut self) -> Result<Oid, Error> {
```

Comment:
Oh one other thing I thought of, I think this should take `self` right to avoid a double-free from calling this twice?

Reply:
I seems to be understand that you mean but I require some hint how to do that right.

Reply:
Oh you can just change this method from `&mut self` to `mut self`

Reply:
Ok, thanks. It seems the borrowing methodology in Rust is much unlike the things I'v ever deal in C and Haskell :)